pkg/controller/projectreference/projectreference_controller.go

```diff
@@ -0,0 +1,116 @@
+	return &ReconcileProjectReference{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("projectreference-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ProjectReference
+	err = c.Watch(&source.Kind{Type: &gcpv1alpha1.ProjectReference{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner ProjectReference
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &gcpv1alpha1.ProjectReference{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileProjectReference implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileProjectReference{}
+
+// ReconcileProjectReference reconciles a ProjectReference object
+type ReconcileProjectReference struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ProjectReference object and makes changes based on the state read
+// and what is in the ProjectReference.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileProjectReference) Reconcile(request reconcile.Request) (result reconcile.Result, err error) {
+	result = reconcile.Result{}
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ProjectReference")
+
+	instance, err := r.getProjectReference(request)
+	if (instance == nil) || (err != nil) {
+		return
+	}
+
+	if instance.Spec.GCPProjectID == "" {
+		err = r.updateProjectName(instance) //this will update the spec, resulting in another run in the loop, hence return
+		return
+	}
+
+	return
+}
+
+func (r *ReconcileProjectReference) getProjectReference(request reconcile.Request) (*gcpv1alpha1.ProjectReference, error) {
+	instance := &gcpv1alpha1.ProjectReference{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return nil, nil
```

Comment:
Instead of returning nil  we should return an empty pointer or object. 

Return` &gcpv1alpha1.ProjectReference{}`

Reply:
wanted to do that, but can't compare two of those with `==`

Reply:
changed to using cmp.Equal, which seems to be able to compare those.